Assets/BossRoom/Scripts/Client/UI/Lobby/JoinLobbyUI.cs

```diff
@@ -0,0 +1,172 @@
+
+        private UpdateRunner m_UpdateRunner;
+
+        private void Awake()
+        {
+            m_LobbyPanelPrototype.gameObject.SetActive(false);
+        }
+
+        private void OnDisable()
+        {
+            if (m_UpdateRunner != null)
+            {
+                m_UpdateRunner.Unsubscribe(PeriodicRefresh);
+            }
+        }
+
+        [Inject]
+        private void InjectDependencies(IInstanceResolver container, LobbyUIMediator lobbyUIMediator, LobbyServiceData lobbyServiceData, UpdateRunner updateRunner)
+        {
+            m_Container = container;
+            m_LobbyUIMediator = lobbyUIMediator;
+            m_LobbyServiceData = lobbyServiceData;
+            m_UpdateRunner = updateRunner;
+
+            m_UpdateRunner.Subscribe(PeriodicRefresh, 10f);
+            BeginObserving(m_LobbyServiceData);
+        }
+
+        public void OnJoinButtonPressed()
+        {
+            m_LobbyUIMediator.JoinLobbyWithCodeRequest(m_JoinCodeField.text.ToUpper());
+        }
+
+        private void PeriodicRefresh(float _)
+        {
+            m_LobbyUIMediator.QueryLobbiesRequest(false);
+        }
+
+        public void OnRefresh()
+        {
+            m_LobbyUIMediator.QueryLobbiesRequest(true);
+        }
+
+        protected override void UpdateObserver(LobbyServiceData observed)
+        {
+            base.UpdateObserver(observed);
+
+            ///Check for new entries, We take CurrentLobbies as the source of truth
+            var previousKeys = new List<string>(m_LobbyButtons.Keys);
+
+            foreach (var codeLobby in observed.CurrentLobbies)
+            {
+                var lobbyCodeKey = codeLobby.Key;
+                var lobbyData = codeLobby.Value;
+
+                if (!m_LobbyButtons.ContainsKey(lobbyCodeKey))
+                {
+                    if (CanDisplay(lobbyData))
+                    {
+                        CreateLobbyPanel(lobbyCodeKey, lobbyData);
+                    }
+                }
+                else
+                {
+                    if (CanDisplay(lobbyData))
+                    {
+                        UpdateLobbyButton(lobbyCodeKey, lobbyData);
+                    }
+                    else
+                    {
+                        RemoveLobbyButton(lobbyData);
+                    }
+                }
+
+                previousKeys.Remove(lobbyCodeKey);
+            }
+
+            foreach (var key in previousKeys)
+                // Need to remove any lobbies from the list that no longer exist.
+            {
+                RemoveLobbyButton(m_LocalLobby[key]);
+            }
+        }
+
+        public void OnQuickJoinClicked()
+        {
+            m_LobbyUIMediator.QuickJoinRequest();
+        }
+
+        private bool CanDisplay(LocalLobby lobby)
+        {
+            return !lobby.Private && lobby.LobbyUsers.Count != lobby.MaxPlayerCount;
```

Comment:
!lobby.Private
Wait what? It's up to clients to filter private lobbies?? How is that private?
If I have pagination with max 16 items and half of those are private, will I just be able to display 8 items? Will my pagination size vary depending on the number of private lobbies in them?

Reply:
And actually, why do we have to do lobby.LobbyUsers.Count != lobby.MaxPlayerCount; as well? This is all data provided by lobby, they should be able to filter this on their side. Right now, that's just consuming bandwidth and costing users more $$ by wasting returned data with useless entries.

Reply:
(I realize this isn't feedback for you Phil, but if you haven't already, we might want to checkin with the lobby team on these)

Reply:
Addressed with custom Order and Filter fields in LobbyAPIInterface